src/main/java/io/jenkins/docker/client/DockerAPI.java

```diff
@@ -205,24 +224,35 @@ public ListBoxModel doFillCredentialsIdItems(@AncestorInPath ItemGroup context,
             if (!ac.hasPermission(Jenkins.ADMINISTER)) {
                 return new StandardListBoxModel().includeCurrentValue(value);
             }
-
             return new StandardListBoxModel().includeAs(
                     ACL.SYSTEM, context, DockerServerCredentials.class,
                     Collections.<DomainRequirement>emptyList());
         }
 
+        public FormValidation doCheckConnectionTimeout(@QueryParameter String value) {
+            return FormValidation.validateNonNegativeInteger(value);
+        }
+
+        public FormValidation doCheckReadTimeout(@QueryParameter String value) {
+            return FormValidation.validateNonNegativeInteger(value);
+        }
+
         public FormValidation doTestConnection(
                 @QueryParameter String uri,
                 @QueryParameter String credentialsId,
                 @QueryParameter String apiVersion,
-                @QueryParameter Integer connectTimeout
-        ) throws IOException, ServletException, DockerException {
+                @QueryParameter int connectTimeout,
+                @QueryParameter int readTimeout
+        ) {
             try {
-                DockerClient dc = new DockerAPI(new DockerServerEndpoint(uri, credentialsId), connectTimeout, apiVersion, null)
-                        .getClient();
-                Version verResult = dc.versionCmd().exec();
-
-                return FormValidation.ok("Version = " + verResult.getVersion() + ", API Version = " + verResult.getApiVersion());
+                final DockerServerEndpoint dsep = new DockerServerEndpoint(uri, credentialsId);
+                final DockerAPI dapi = new DockerAPI(dsep, connectTimeout, readTimeout, apiVersion, null);
+                final DockerClient dc = dapi.getClient();
+                final VersionCmd vc = dc.versionCmd();
+                final Version v = vc.exec();
+                final String actualVersion = v.getVersion();
+                final String actualApiVersion = v.getApiVersion();
+                return FormValidation.ok("Version = " + actualVersion + ", API Version = " + actualApiVersion);
```

Comment:
Splitting the test code up so that we only get one method call per line means that, in the event of a failure, the exception trace will tell us exactly where the problem came from.
e.g. If everything is on one line and we get a NullPointerException then it's impossible to see which bit failed.